internal/kubenurse/handler.go

```diff
@@ -0,0 +1,70 @@
+package kubenurse
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/postfinance/kubenurse/internal/kubediscovery"
+)
+
+func (s *Server) readyHandler() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+
+		if s.ready {
+			w.WriteHeader(http.StatusOK)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	}
+}
+
+func (s *Server) aliveHandler() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		type Output struct {
+			Hostname   string              `json:"hostname"`
+			Headers    map[string][]string `json:"headers"`
+			UserAgent  string              `json:"user_agent"`
+			RequestURI string              `json:"request_uri"`
+			RemoteAddr string              `json:"remote_addr"`
+
+			// checker.Result
+			APIServerDirect string `json:"api_server_direct"`
+			APIServerDNS    string `json:"api_server_dns"`
+			MeIngress       string `json:"me_ingress"`
+			MeService       string `json:"me_service"`
```

Comment:
would be better to have the checker result embedded as part of the struct.

something such as
```
type Output struct {
serviceCheck.Result
Hostname ...
Headers...
}
```

Reply:
Good catch! This code was also in the old codebase and I didn't change it. Will do now...